diff -rc linux-86.orig/copt/copt.c linux-86/copt/copt.c
*** linux-86.orig/copt/copt.c	Tue Feb 18 21:31:34 1997
--- linux-86/copt/copt.c	Tue Apr 22 14:11:05 1997
***************
*** 303,314 ****
  }
  
  
- 
- /*
-  * Eval an expression into an integer number
-  */
- static long eval(char *str, int len)
- {
  #define NO_OP  0
  #define ADD_OP 1
  #define SUB_OP 2
--- 303,308 ----
***************
*** 317,330 ****
  #define SHL_OP 5
  #define SHR_OP 6
  
!   char *oldcp, *cp, c;
!   long retval = 0;
!   long num = 0;
!   int sign = 1;
!   int base = 10;
!   int state = 0;
!   int op = NO_OP;
!   int i, varnum;
  
    /* Apply operation to current numeric value */
    static void doretval(void)
--- 311,321 ----
  #define SHL_OP 5
  #define SHR_OP 6
  
!   static int op = NO_OP;
!   static long retval = 0;
!   static long num = 0;
!   static int sign = 1;
!   static int base = 10;
  
    /* Apply operation to current numeric value */
    static void doretval(void)
***************
*** 350,355 ****
--- 341,363 ----
  	sign = 1;
  	base = 10;
    }
+ 
+ 
+ /*
+  * Eval an expression into an integer number
+  */
+ static long eval(char *str, int len)
+ {
+ 
+   char *oldcp, *cp, c;
+   int state = 0;
+   int i, varnum;
+ 
+   op = NO_OP;
+   retval = 0;
+   num = 0;
+   sign = 1;
+   base = 10;
  
    /* Scan through whole string and decode it */
    for (cp = str, i = 0; *cp && i < len; cp++, i++) {
